src/components/player.py

```diff
@@ -2,16 +2,43 @@
 from .abstracts import AbstractActor, Base
 from .types import Args, Kwargs
 from .keyboardcontroller import KeyboardController
+from .equipable import Gun
+from .types import (
+    Args,
+    Kwargs,
+    EdgeBehavior,
+)  # Imported this because Player __init__ references EdgeBehavior
 
 
 class Player(AbstractActor):
     U = 16
     V = 64
     w = AbstractActor.BASE_BLOCK * 2
     h = AbstractActor.BASE_BLOCK * 2
+    weapon_slot = Gun()
+
+    def __init__(
+        self,
+        *args: Args,
+        controller: Any = None,  # figure out how to change this Any to AbstractController
+        x: int = 0,
+        y: int = 0,
+        speed: int = 0,
+        edge_behavior: EdgeBehavior = EdgeBehavior.stop,
+        **kwargs: Kwargs
+    ) -> None:
+        self.controller = controller
+        self.x = x
+        self.y = y
+        self.speed = speed
+        self.controller.register(self)
+        # todo: maybe add different config for looping screen
+        # or leaving screen
+        self.edge_behavior: EdgeBehavior = edge_behavior
```

Comment:
`self.weapon_slot: AbstractItem` 
would go here and in the `AbstractActor` definition of `__init__`